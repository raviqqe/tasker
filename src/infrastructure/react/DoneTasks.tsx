import { defaultImport } from "default-import";
import defaultInfiniteScroll, {
  type Props,
} from "react-infinite-scroll-component";
import { styled } from "@linaria/react";
import { type ITask } from "../../domain/task.js";
import { Loader } from "./Loader.js";
import { Task } from "./Task.js";


const InfiniteScroll = defaultImport(defaultInfiniteScroll);

const doneTasksContainerId = "done-tasks-container";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  overflow: auto;
`;

const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StyledInfiniteScroll = styled((props: Props) => (
  <InfiniteScroll {...props} />
))`
  display: flex;
  flex-direction: column;
  padding: 1em 0.5em;
`;

const StyledTask = styled(Task)`
  margin: 0.5em;
`;

export interface IProps {
  doneTasks: ITask[] | null;
  listMoreDoneTasks: () => Promise<void>;
}

export const DoneTasks = ({
  doneTasks,
  listMoreDoneTasks,
}: IProps): JSX.Element =>
  doneTasks ? (
    <Container id={doneTasksContainerId}>
      <StyledInfiniteScroll
        dataLength={doneTasks.length}
        hasMore
        loader={null}
        next={listMoreDoneTasks}
        scrollableTarget={doneTasksContainerId}
      >
        {doneTasks.map((task: ITask) => (
          <StyledTask key={task.id} task={task} />
        ))}
      </StyledInfiniteScroll>
    </Container>
  ) : (
    <LoaderContainer>
      <Loader />
    </LoaderContainer>
  );
