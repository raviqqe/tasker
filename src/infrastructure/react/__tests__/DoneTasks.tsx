import React from "react";
import { create } from "react-test-renderer";
import { DoneTasks } from "../DoneTasks";

it("renders", () => {
  expect(
    create(
      <DoneTasks doneTasks={[]} listMoreDoneTasks={async () => undefined} />
    ).toJSON()
  ).toMatchSnapshot();
});
