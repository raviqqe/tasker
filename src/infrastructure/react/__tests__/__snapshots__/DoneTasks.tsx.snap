// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-eCImvq tPbwh"
  id="done-tasks-container"
>
  <div
    className="infinite-scroll-component__outerdiv"
    style={Object {}}
  >
    <div
      className="infinite-scroll-component sc-gKckTs fWORFv"
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "height": "auto",
          "overflow": "auto",
        }
      }
    />
  </div>
</div>
`;
